### PR TITLE
caveats: Remove extra blank line between `keg_only` and other caveats

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- This was pointed out by Carlo in https://github.com/Homebrew/brew/issues/14925#issuecomment-1473303922, it annoyed me the moment I saw it too, and it was a very quick fix.
- The caveats items are joined by "\n" before being displayed, so the extra `\n` here when generating the `keg_only_text` meant that there were two blank lines which looked weird. This `keg_only_text` method is only used here so it doesn't affect anywhere else we show the `keg_only` message, if there are any other places.

-----

Before:

```
$ HOMEBREW_NO_INSTALL_FROM_API=1 brew info socket_vmnet
[...]
==> Caveats
socket_vmnet requires root privileges so you will need to run
  `sudo /opt/homebrew/opt/socket_vmnet/socket_vmnet` or `sudo brew services start socket_vmnet`.
You should be certain that you trust any software you grant root privileges.

socket_vmnet is keg-only, which means it was not symlinked into /opt/homebrew,
because /opt/homebrew/bin is often writable by a non-admin user.


To restart socket_vmnet after an upgrade:
  sudo brew services restart socket_vmnet
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet --vmnet-gateway=192.168.105.1 /opt/homebrew/var/run/socket_vmnet
```

After:

```
$ HOMEBREW_NO_INSTALL_FROM_API=1 brew info socket_vmnet
[...]
==> Caveats
socket_vmnet requires root privileges so you will need to run
  `sudo /opt/homebrew/opt/socket_vmnet/socket_vmnet` or `sudo brew services start socket_vmnet`.
You should be certain that you trust any software you grant root privileges.

socket_vmnet is keg-only, which means it was not symlinked into /opt/homebrew,
because /opt/homebrew/bin is often writable by a non-admin user.

To restart socket_vmnet after an upgrade:
  sudo brew services restart socket_vmnet
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet --vmnet-gateway=192.168.105.1 /opt/homebrew/var/run/socket_vmnet
```

### CODE REVIEW

**Review by carlocab — APPROVED** [review]
omg i love this

**Comment by carlocab** [comment]
Oh, and happy #15000! 🎉

File: Library/Homebrew/caveats.rb
Comment (Rylan12) [inline Library/Homebrew/caveats.rb:94]: If neither of these conditionals are met, we don't get the newline added. Should we add something like this to make sure it's consistently done?

```ruby
if !(formula.lib/"pkgconfig").directory? && !(formula.share/"pkgconfig").directory?
  s << "\n"
end
```

(there's also a spot above that's similar)

**Review by Rylan12 — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/caveats.rb
Comment (issyl0) [reply Library/Homebrew/caveats.rb:94]: I did this a different way, but yes.

**Review by Rylan12 — APPROVED** [review]
Looks good!

**Comment by issyl0** [comment]
I made the test slightly better, but the actual code change is the same.

**Review by Rylan12 — APPROVED** [review]

**Comment by issyl0** [comment]
This isn't strictly "critical" but it has been reviewed by multiple pairs of eyes. And the merge button isn't blocked because the "review" CI check isn't required anymore (because it didn't work with the merge queue). I've asked in Slack for clarification on whether I _should_ merge this as-is without the "critical" label - it has had multiple reviews from other maintainers, after all - or add the "critical" label to get all the CI jobs to pass, or wait until the review period (which is no longer as strictly enforced) is up.

**Comment by Rylan12** [comment]
👍 with me to mark as `critical` and merge. Agree it's not clear what the consensus is on the `critical` policy but I don't think this change is controversial at all

**Pull request merged**
